### PR TITLE
Replace ManagementFactoryHelper with ManagementFactory for java11

### PR DESCRIPTION
**Pull request by yyongycy**
### What changes are proposed in this pull request?

Replace ManagementFactoryHelper with ManagementFactory for java11

### Why are the changes needed?
Java 11 doesn't support to directly call ManagementFactoryHelper, so it is necessary to find an alternative one in java11.

### Does this PR introduce any user facing changes?

NA

### CODE REVIEW

**Comment by Xenorith** [comment]
were you able to run some code with this change in a java 11 environment?

for example, you could use the docker image at https://hub.docker.com/layers/alluxio/alluxio-maven/0.0.8-jdk11/images/sha256-84e7253045d53cc2ebfce98e3d6710b1d446e9724115df3dbf4bf41935b25d7c?context=explore as a way to run some of the same maven commands as the PR checks

**Comment by yyongycy** [comment]
> were you able to run some code with this change in a java 11 environment?
> 
> for example, you could use the docker image at https://hub.docker.com/layers/alluxio/alluxio-maven/0.0.8-jdk11/images/sha256-84e7253045d53cc2ebfce98e3d6710b1d446e9724115df3dbf4bf41935b25d7c?context=explore as a way to run some of the same maven commands as the PR checks

So still no chance to compile, am I right?

I have a small program to try with java11:


import java.lang.management.BufferPoolMXBean;
import java.lang.management.ManagementFactory;
import java.lang.management.MemoryPoolMXBean;
import java.nio.ByteBuffer;

public class JavaExceptionTest {

  public static void main(String[] args) {
    // BufferPoolMXBean bean = getDirectBufferPool();
    // MemoryPoolMXBean bean = getDirectBufferPool();
    // System.out.println("Got bean:" + bean);
    BufferPoolMXBean memoryPoolMXBean = getDirectBuffer();
    System.out.println("Got bean:" + memoryPoolMXBean);
    System.out.println(String.format("DirectMem used: %d", memoryPoolMXBean.getMemoryUsed()));
    ByteBuffer.allocateDirect(5);
    System.out.println(String.format("DirectMem used: %d", memoryPoolMXBean.getMemoryUsed()));
  }

  private static MemoryPoolMXBean getDirectBufferPool() {
  // private static BufferPoolMXBean getDirectBufferPool() {
    for (MemoryPoolMXBean bufferPoolMXBean
        : java.lang.management.ManagementFactory.getMemoryPoolMXBeans()) {
    // for (BufferPoolMXBean bufferPoolMXBean
        // : sun.management.ManagementFactoryHelper.getBufferPoolMXBeans()) {
      System.out.println(String.format("found pool: %s %s",
          bufferPoolMXBean, bufferPoolMXBean.getName()));
      for (String str : bufferPoolMXBean.getMemoryManagerNames()) {
        System.out.println(String.format("the MMNme: %s", str));
      }
      if (bufferPoolMXBean.getName().equals("direct")) {
        return bufferPoolMXBean;
      }
    }
    return null;
  }

  private static BufferPoolMXBean getDirectBuffer() {
    // private static BufferPoolMXBean getDirectBufferPool() {
    for (BufferPoolMXBean bufferPoolMXBean
        : java.lang.management.ManagementFactory.getPlatformMXBeans(BufferPoolMXBean.class)) {
      System.out.println(String.format("found pool: %s %s",
          bufferPoolMXBean, bufferPoolMXBean.getName()));
      if (bufferPoolMXBean.getName().equals("direct")) {
        return bufferPoolMXBean;
      }
    }
    return null;
  }
}

**Comment by yyongycy** [comment]
Following is the commands:

alluxio@MacBook-Pro-3 java % java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
alluxio@MacBook-Pro-3 java % javac JavaExceptionTest.java
alluxio@MacBook-Pro-3 java % java JavaExceptionTest
found pool: sun.management.ManagementFactoryHelper$1@13fee20c mapped
found pool: sun.management.ManagementFactoryHelper$1@4e04a765 direct
Got bean:sun.management.ManagementFactoryHelper$1@4e04a765
DirectMem used: 0
DirectMem used: 5

**Comment by Xenorith** [comment]
> > were you able to run some code with this change in a java 11 environment?
> > for example, you could use the docker image at https://hub.docker.com/layers/alluxio/alluxio-maven/0.0.8-jdk11/images/sha256-84e7253045d53cc2ebfce98e3d6710b1d446e9724115df3dbf4bf41935b25d7c?context=explore as a way to run some of the same maven commands as the PR checks
> 
> So still no chance to compile, am I right?
> 

You would be able to compile the code within the docker container with java 8 and then run the maven commands with java 11. This follows the way we build the code and run it on a different java version since we always build with java 8 and only test compatibility with other java versions at runtime.

File: dev/jenkins/run_docker.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ function main {
   fi
   if [ -z "${ALLUXIO_DOCKER_IMAGE}" ]
   then
-    ALLUXIO_DOCKER_IMAGE="alluxio/alluxio-maven:0.0.8-jdk8"
+    ALLUXIO_DOCKER_IMAGE="alluxio/alluxio-maven:0.0.8-jdk11"
```
Comment (yyongycy) [inline dev/jenkins/run_docker.sh:26]: @Xenorith  this is to try with jdk11,

**Comment by yyongycy** [comment]
image
Once all tests pass, I will revert this change.

**Comment by yyongycy** [comment]
> image Once all tests pass, I will revert this change.

image

**Comment by yyongycy** [comment]
> > > were you able to run some code with this change in a java 11 environment?
> > > for example, you could use the docker image at https://hub.docker.com/layers/alluxio/alluxio-maven/0.0.8-jdk11/images/sha256-84e7253045d53cc2ebfce98e3d6710b1d446e9724115df3dbf4bf41935b25d7c?context=explore as a way to run some of the same maven commands as the PR checks
> > 
> > 
> > So still no chance to compile, am I right?
> 
> You would be able to compile the code within the docker container with java 8 and then run the maven commands with java 11. This follows the way we build the code and run it on a different java version since we always build with java 8 and only test compatibility with other java versions at runtime.

Verified with the container,

**Review by Xenorith — APPROVED** [review]

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Pull request merged**